### PR TITLE
feat: Add job-based Slack notification docs

### PR DESCRIPTION
**Pull request by jweingarten**
Documentation for https://github.com/screwdriver-cd/notifications-slack/pull/24. Merge after implementation of feature was merged.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — APPROVED** [review]

File: docs/user-guide/metadata.md [hunk 1]
```diff
@@ -224,3 +225,20 @@ jobs:
 
 Result:
 ![notification-meta](./assets/notification-meta.png)
+
+#### Job based
+*Note*: Job based slack notification meta data will overwrite the Basic notification message.
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
*Note*: Job-based Slack notification meta data will overwrite the basic notification message. 
```

File: docs/user-guide/metadata.md [hunk 2]
```diff
@@ -224,3 +225,20 @@ jobs:
 
 Result:
 ![notification-meta](./assets/notification-meta.png)
+
+#### Job based
+*Note*: Job based slack notification meta data will overwrite the Basic notification message. 
+
+Structure of meta variable is `notification.slack.<jobname>.message` replacing `<jobname>` with the name of the SD job.
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
Structure of meta variable is `notification.slack.<jobname>.message`, replacing `<jobname>` with the name of the Screwdriver job.
```

File: docs/user-guide/metadata.md [hunk 3]
```diff
@@ -224,3 +225,20 @@ jobs:
...
+
+Example screwdriver.yaml notifying with specific slack message for job `slack-notification-test`:
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
Example screwdriver.yaml notifying with specific Slack message for job `slack-notification-test`:
```

File: docs/user-guide/metadata.md [hunk 4]
```diff
@@ -224,3 +225,20 @@ jobs:
...
+```yaml
+jobs:
+  main:
+    steps:
+      - meta: |
+          meta set notification.slack.slack-notification-test.message "<@yoshwata> Hello!!"
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
          meta set notification.slack.slack-notification-test.message "<@yoshwata> Hello Meta!"
```

changing message to match image

**Review by tkyi — APPROVED** [review]
👍

**Pull request merged**